src/test/java/org/prebid/server/bidder/somoaudience/SomoaudienceBidderTest.java

```diff
@@ -76,35 +80,96 @@ public void makeHttpRequestsShouldReturnHttpRequestWithCorrectBodyHeadersAndMeth
+                        .user(User.builder().ext(mapper.valueToTree(ExtUser.of(null, "consent", null))).build())
+                        .regs(Regs.of(0, mapper.valueToTree(ExtRegs.of(1))))
+                        .ext(mapper.valueToTree(SomoaudienceReqExt.of("hb_pbs_1.0.0")))
+                        .device(Device.builder().ua("User Agent").ip("ip").dnt(1).language("en").build())
                         .build()
         ));
     }
 
+    @Test
+    public void makeHttpRequestsShouldReturnCorrectRequestBodyAndUri() throws IOException {
+        // given
+        final BidRequest bidRequest = BidRequest.builder().imp(asList(
+                Imp.builder()
+                        .banner(Banner.builder().build())
+                        .ext(mapper.valueToTree(ExtPrebid.of(
+                                null, ExtImpSomoaudience.of("placement1", BigDecimal.valueOf(1.5)))))
+                        .build(),
+                Imp.builder()
+                        .banner(Banner.builder().build())
+                        .ext(mapper.valueToTree(ExtPrebid.of(
+                                null, ExtImpSomoaudience.of("placement2", BigDecimal.valueOf(1.33)))))
+                        .build(),
+                Imp.builder()
+                        .video(Video.builder().build())
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpSomoaudience.of("placement3", null))))
+                        .build(),
+                Imp.builder()
+                        .xNative(Native.builder().build())
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpSomoaudience.of("placement4", null))))
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = somoaudienceBidder.makeHttpRequests(bidRequest);
+
+        // then
+        final String expectedBannersString = mapper.writeValueAsString(BidRequest.builder()
+                .imp(asList(
+                        Imp.builder().banner(Banner.builder().build()).bidfloor(1.5F).build(),
+                        Imp.builder().banner(Banner.builder().build()).bidfloor(1.33F).build()))
+                .ext(mapper.valueToTree(SomoaudienceReqExt.of("hb_pbs_1.0.0")))
+                .build());
+        final String expectedVideoSting = mapper.writeValueAsString(BidRequest.builder()
+                .imp(singletonList(Imp.builder().video(Video.builder().build()).build()))
+                .ext(mapper.valueToTree(SomoaudienceReqExt.of("hb_pbs_1.0.0")))
+                .build());
+        final String expectedNativeString = mapper.writeValueAsString(BidRequest.builder()
+                .imp(singletonList(Imp.builder().xNative(Native.builder().build()).build()))
+                .ext(mapper.valueToTree(SomoaudienceReqExt.of("hb_pbs_1.0.0")))
+                .build());
+
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(3).element(0)
```

Comment:
Same 3 checks could be combined in one `assertThat` invocation using `extracting(HttpRequest::getUri, HttpRequest::getBody)` and further `containsOnly` check against a list of tuples